sys/filesys_windows.go

```diff
@@ -66,7 +66,7 @@ func mkdirall(path string, adminAndLocalSystem bool) error {
 		return &os.PathError{
 			Op:   "mkdir",
 			Path: path,
-			Err:  syscall.ENOTDIR,
+			Err:  windows.ERROR_PATH_NOT_FOUND,
```

Comment:
After some discussion on https://github.com/moby/moby/pull/40645#discussion_r389694799, I'll change this one back; opening a PR to revert this one